dhcpcd: 7.2.3 -> 8.0.2

dtzWill opened the pull request:
###### Motivation for this change

https://roy.marples.name/blog/dhcpcd-8-0-0-released
https://roy.marples.name/blog/dhcpcd-8-0-1-released
https://roy.marples.name/blog/dhcpcd-8-0-2-released

Haven't checked the tests yet, should definitely check before proceeding
:).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
dhcpcd: 7.2.3 -> 8.0.1

https://roy.marples.name/blog/dhcpcd-8-0-0-released
https://roy.marples.name/blog/dhcpcd-8-0-1-released

Ma27 commented:
As I have a non-trivial dhcpcd configuration on my main workstation, I'd like to test this before having this merged (if I have sufficient time, I'd do this tomorrow :) )

Ma27 reviewed (commented):
Just tested my personal setup with `dhcpcd` 8.0.1. In contrast to a traditional setup it doesn't set a default gateway and a default DNS server. Instead I use a local resolver and an always-on VPN and set routes to the VPN endpoint in an exit hook (see `dhcpcd-run-hooks(8)` for further reference). I just used this change for half a day and tested several scenarios including reboots and network changes and I didn't encounter any regressions. Also tested `nixos.tests.networking.scripted.privacy.x86_64-linux`, `nixos.tests.networking.scripted.dhcpSimple.x86_64-linux` and `nixos.tests.networking.scripted.macvlan.x86_64-linux`.

However it would be awesome to have some more people who could test this :)

dtzWill committed:
dhcpcd: 8.0.1 -> 8.0.2

https://roy.marples.name/blog/dhcpcd-8-0-2-released

> I don't anticipate any more releases for a while as this is looking
really good now!

dtzWill commented:
Upstream notes 8.0.2 is last expected release for a bit, since things are apparently looking good now (paraphrased from release notes linked above :)).

mmahut commented:
@GrahamcOfBorg test networking

mmahut commented:
I did test it a little, looks to be fine.
